### PR TITLE
Added StackingPenaltyRule

### PR DESCRIPTION
**Pull request by Shirkanesi**
Added option to create assessment-buttons which will apply their penalty on every click.

A new PanaltyRule is introduced for this. Sample configuration:
```json
"penaltyRule": {
  "shortName": "stackingPenalty",
  "penalty": 0.5,
  "maxUses": 4 
}
```
Note: `maxUses` can be left out or explicitly set to `null` to remove the cap for the stackingPenalty.

### CODE REVIEW

**Review by dfuchss — APPROVED** [review]
lgtm.

@YvesKirschner I think that might be usable in the future :)

**Comment by Shirkanesi** [comment]
I'll also add an option to cap the maximum amount of deductions for this penalty later.

**Comment by dfuchss** [comment]
@Shirkanesi could you look at my last commit to the branch? Is that ok for you?

**Comment by dfuchss** [comment]
> I'll also add an option to cap the maximum amount of deductions for this penalty later.

ok then I'll wait with the merge

**Comment by Shirkanesi** [comment]
> @Shirkanesi could you look at my last commit to the branch? Is that ok for you?

When you are happy with coupling the PenaltyRules that close with the JsonNode...
It's ok for me but not happy.

**Comment by dfuchss** [comment]
> > @Shirkanesi could you look at my last commit to the branch? Is that ok for you?
> 
> When you are happy with coupling the PenaltyRules that close with the JsonNode... It's ok for me but not perfectly happy.

If you have a better idea, I would be happy. But for now I think it's better that the deserialization is located within the class and not inplace in the enum constructor.

File: bundles/edu.kit.kastel.sdq.eclipse.common.core/src/edu/kit/kastel/sdq/eclipse/common/core/model/StackingPenaltyRule.java [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+public class StackingPenaltyRule extends PenaltyRule {
+
+	private static final String DISPLAY_NAME = "Stacking Penalty";
+	public static final String SHORT_NAME = "stackingPenalty";
+
+	// Penalty stored with decimal-point shifted one to the right (make sure no
+	// rounding issues happen)
+	private int penalty;
+
+	@Nullable
```
Comment (dfuchss) [inline bundles/edu.kit.kastel.sdq.eclipse.common.core/src/edu/kit/kastel/sdq/eclipse/common/core/model/StackingPenaltyRule.java:21]: I'm not sure about the Nullable annotation from JGit.

**Review by dfuchss — APPROVED** [review]
In general this lgtm . Only one comment :)

File: bundles/edu.kit.kastel.sdq.eclipse.common.core/src/edu/kit/kastel/sdq/eclipse/common/core/model/StackingPenaltyRule.java (hunk 1 above)
Comment (Shirkanesi) [reply bundles/edu.kit.kastel.sdq.eclipse.common.core/src/edu/kit/kastel/sdq/eclipse/common/core/model/StackingPenaltyRule.java]: I removed it.

**Review by dfuchss — APPROVED** [review]
Lgtm.

**Pull request merged**